### ISSUE TITLE
Enable Custom esbuild config

### ISSUE THREAD

**Issue by skorfmann** [winglang/wing#3410]
### Feature Spec

Wing now supports customizing the esbuild config for bundling inflight functions. This enables bundling functions with custom requirements, such as WebAsssembly modules or using custom esbuild plugins.

### Use Cases

- enable usage of custom esbuild plugins
- enable usage of custom loaders for  WASM modules or similar formats 

### Component

Compiler

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by skorfmann** [comment]
Just ran into the error described here https://answers.netlify.com/t/lambda-functions-fastify-error-cannot-find-module-string-decoder/40219/12 in one of the examples

**Comment by skyrpex** [comment]
A related use case is running `cloud.Service` during `sim` and accessing modules such as `vite`, where the default esbuild bundling config fails.

**Comment by eladb** [comment]
> ```
> external_node_modules = ["readable-stream"]
> ```


@skorfmann would it make sense to always build with this setting when bundling for aws lambda?

**Comment by skyrpex** [comment]
Dependencies such as the `fsevents` dependency should also never be bundled AFAIK.

**Comment by skorfmann** [comment]
This would help with using jsx in the Typescript SDK as well, e.g. something like this https://github.com/winglang/examples/pull/45

**Comment by Chriscbr** [comment]
See also the bundling options offered by CDK here: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_nodejs.BundlingOptions.html

One design option could be to have some way to configure bundling on a per `cloud.Function` basis:

```js
new cloud.Function(handler,
  env: {
    "KEY": "VALUE",
  },
  bundling: {
    external: ["fsevents"],
    minify: false,
  }
);
```

Another option is to have some way to configure bundling app-wide:

```js
// inside the platform or in Wing, both can be supported
std.bundling.configure(
  external: @app.target == "aws" ? ["fsevents", "aws-sdk"] : ["fs-events"],
  minify: false,
);
```

When we expose an API for serializing an inflight into a string, bundling options would also be exposed for that naturally (this would be the lowest level).

@eladb thoughts?

**Comment by eladb** [comment]
Good idea - I think it might be sufficient as an initial step just support this at the app level.

Only thing, I'd try to avoid a global and expose it on via the `@app`:

```js
@app.bundling.configure(...)
```
